honeybee_grasshopper_radiance/src/HB Extract HDR.py

```diff
@@ -0,0 +1,218 @@
+# along with Honeybee; If not, see <http://www.gnu.org/licenses/>.
+# 
+# @license GPL-3.0+ <http://spdx.org/licenses/GPL-3.0+>
+
+"""
+Interpolate or extrapolate a High Dynamic Range (HDR) image file from another
+HDR image file.
+_
+Recommended use is to extract 180 FOV (-vh 180 -vv 180) angular or hemispherical
+HDR images from a 360 FOV (-vh 360 -vv 360) angular HDR image. Alternatively
+conversions between 180 FOV angular and hemispherical HDR images can be made.
+-
+
+    Args:
+        _view: A view to interpolate or extrapolate. Use "HB View" to create a
+            view.
+        _resolution_: An integer for the dimension of the output image in pixels.
+            If extracting a 180 FOV angular or hemispherical HDR image from a
+            360 FOV HDR image, the default resolution is 1/3 of the resolution of
+            _hdr. If converting between 180 FOV angular or hemispherical HDR 
+            images, the default resolution is that of _hdr.
+        _hdr: Path to a High Dynamic Range (HDR) image file from which to
+            interpolate or extrapolate a view.
+
+    Returns:
+        hdr: Path to the resulting HDR image file.
+"""
+
+ghenv.Component.Name = 'HB Extract HDR'
+ghenv.Component.NickName = 'ExtractHDR'
+ghenv.Component.Message = '1.3.0'
+ghenv.Component.Category = 'HB-Radiance'
+ghenv.Component.SubCategory = '4 :: Results'
+ghenv.Component.AdditionalHelpFromDocStrings = '3'
+
+import os
+import subprocess
+
+try:  # import honeybee_radiance_command dependencies
+    from honeybee_radiance_command.pinterp import Pinterp
+    from honeybee_radiance_command.ra_xyze import Ra_xyze
+except ImportError as e:
+    raise ImportError('\nFailed to import honeybee_radiance_command:\n\t{}'.format(e))
+
+try:  # import honeybee_radiance dependencies
+    from honeybee_radiance.config import folders as rad_folders
+    from honeybee_radiance.view import View
+except ImportError as e:
+    raise ImportError('\nFailed to import honeybee_radiance:\n\t{}'.format(e))
+
+try:  # import ladybug_rhino dependencies
+    from ladybug_rhino.grasshopper import all_required_inputs, give_warning
+except ImportError as e:
+    raise ImportError('\nFailed to import ladybug_rhino:\n\t{}'.format(e))
+
+# check the Radiance date of the installed radiance
+try:  # import lbt_recipes dependencies
+    from lbt_recipes.version import check_radiance_date
+except ImportError as e:
+    raise ImportError('\nFailed to import lbt_recipes:\n\t{}'.format(e))
+check_radiance_date()
+
+
+def check_view_hdr(hdr_path):
+    """Check if the header of the HDR image contains a view (VIEW=).
+    
+    A ValueError is raised if the image does not contain a valid view.
+    A ValueError is raised if the view type is not -vta or -vth.
+    
+    Args:
+        hdr_path: The path to an HDR image file.
+    """
+    # set valid view = False. Will later be set to True if a valid view is found
+    valid_view = False
```

Comment:
Do we need this variable? Why not just set `hdr_view_ to None here and then later check if its still None.